miniwob/spaces.py

```diff
@@ -0,0 +1,176 @@
+"""Custom spaces for MiniWoB++."""
+from typing import Any, FrozenSet, Optional, Tuple, Union
+
+import numpy as np
+from gymnasium.spaces import Box
+from gymnasium.spaces.space import Space
+from gymnasium.spaces.text import alphanumeric
+from gymnasium.spaces.utils import flatdim, flatten, flatten_space, unflatten
+from numpy.typing import NDArray
+
+
+MAX_UNICODE_CODEPOINT = 0x10FFFF
+
+
+class Unicode(Space[str]):
```

Comment:
Given that this largely a copy of the `Text` with modifications.
Is there any point to inheriting from `Text`? `Unicode(Text)`? And modifying the respective functions?

Reply:
I have now switched to subclassing `Text`.

I originally had a concern that the `singledispatch` methods (functions that use type-based registration, like `flatten` and `unflatten`) might not work properly with subclasses, but they seem to work fine.

Reply:
Perfect, thanks